### DISCUSSION TITLE
WHEP client as a standalone library?

### DISCUSSION THREAD

**Discussion by rgl** [bluenviron/mediamtx#2416] [Questions]
@aler9, at https://github.com/bluenviron/mediamtx/discussions/1581#discussioncomment-5560607 you've described how to use the WHEP client by copy&paste. Does it make sense to create a standalone library of it? Or maybe collaborate with [whip-whep-js](https://github.com/medooze/whip-whep-js)? Or is it too soon in the WHEP release cycle for doing that?

**Comment by aler9** [comment]
for anyone reading this question: in order to publish with WHIP or read with WHEP with JS, you can either use

* [whip-whep-js](https://github.com/medooze/whip-whep-js), that is the reference WHIP JS implementation
* the [whep page](https://github.com/bluenviron/mediamtx/blob/main/internal/core/webrtc_read_index.html) or the [whip page](https://github.com/bluenviron/mediamtx/blob/main/internal/core/webrtc_publish_index.html) bundled into the server

**Reply by 0xAl3xH** [reply]
@aler9 the whep page and whip page are out of date, is it [this](https://github.com/bluenviron/mediamtx/blob/cf7088a91b6d90d9914007dbacc42b414d503d52/internal/servers/webrtc/read_index.html)?

**Reply by aler9** [reply]
discussions are not up-to-date references, while the README and the code are. Take a look at either the README or the code in order to find out the right links.

**Reply by ZakEspley** [reply]
The links to the WHEP and WHIP page don't work anymore. Is it the same as:

https://github.com/bluenviron/mediamtx/blob/main/internal/servers/webrtc/read_index.html

This is the link you have in the README.

I have tried copying this file and the reader.js file to my nginx root location, but when I try to load the webpage I get 405 errors. Which I presume is related to my NGINX configuration, but it seemed like since browsers can consume WHEP content it would be easy to implement, so maybe it is not an NGINX error.

> for anyone reading this question: in order to publish with WHIP or read with WHEP with JS, you can either use
> 
>     * [whip-whep-js](https://github.com/medooze/whip-whep-js), that is the reference WHIP JS implementation
> 
>     * the [whep page](https://github.com/bluenviron/mediamtx/blob/main/internal/core/webrtc_read_index.html) or the [whip page](https://github.com/bluenviron/mediamtx/blob/main/internal/core/webrtc_publish_index.html) bundled into the server

**Reply by aler9** [reply]
@ZakEspley it's impossible to debug your error without server logs or further details. Open a bug report and provide detailed instructions on how to replicate the issue.
I'm forced to close this discussion since it's outdated and off topic.
